tools/scoring/score.py

```diff
@@ -81,35 +84,51 @@ def add_division_family_info(data,fam):
     return data
 
 
-# Drop any rows that contain benchmarks with status unknown where two otherwise
-# sound solvers disagree on the result
-def removeDisagreements(data):
+def remove_disagreements(data):
     global g_args
-    # First find unsound solvers e.g. those that disagree with the expected
-    # status these solvers are ignored in the following
+
+    # First find and filter out unsound solvers ,i.e., solvers that disagree
+    # with the expected status.
     unsound_solvers = set(data[(data.expected != "starexec-unknown")
                                & (data.result != "starexec-unknown")
                                & (data.result != data.expected)]['solver'])
- 
-    if g_args.log: log("Removing disagreements...")
-    unknown = data[(data.expected == "starexec-unknown")
-                   & (~data.solver.isin(unsound_solvers))
-                   & ((data.result == 'sat')
-                      | (data.result == 'unsat'))]
-    exclude = set()
-    for b,rows in unknown.groupby('benchmark'):
-        res = set(rows.result.tolist())
-        if 'sat' in res and 'unsat' in res:
-            if g_args.log: log("Removing "+b)
-            if g_args.log: log(view(rows))
-            exclude.add(b)
+
+    # Consider only unknown benchmarks that were solved by sound solvers.
+    solved_unknown = data[(data.expected == "starexec-unknown")
+                          & (~data.solver.isin(unsound_solvers))
+                          & ((data.result == 'sat') | (data.result == 'unsat'))]
+
+    # Remove duplicate (benchmark, result) pairs to produces unique
+    # result values for each benchmark.
+    solved_unknown = solved_unknown.drop_duplicates(
+                            subset=['benchmark', 'result'])
+
+    # Group by benchmarks and count the number of results.
+    grouped_results = solved_unknown.groupby('benchmark', as_index=False).agg(
+                            {'result': 'count'})
+
+    # If the number of results is more than one, we have disagreeing solvers, 
+    # i.e., the result column contains 'sat' and 'unsat' for the corresponding
+    # benchmark.
+    disagreements = grouped_results[grouped_results['result'] > 1]
+
+    exclude = set(disagreements.benchmark)
+
+    if g_args.log:
+        log('Found {} disagreements:'.format(len(exclude)))
+        i = 1
+        for b in exclude:
+            log('[{}] {}'.format(i, b))
+            i += 1
+
+    # Exclude benchmarks on which solvers disagree.
```

Comment:
This was optional before, right? (`removeDisagreements`)
Are disagreements always removed now?
Also, it might be good to list them.

Reply:
This is still optional (checked in `process_csv()`). Disagreements are printed in the `if g_args.log` block.

Reply:
FYI - I made removing disagreements optional because my implementation was so slow it was annoying in testing, so I would turn it off whilst testing other bits. As you've removed the efficiency issue it might make sense to remove the option as there doesn't seem to be another use case where we might want to turn it off.

Reply:
Agreed, we can make this non-optional. But we *must* print a list of disagreements if they exist.